ChiModules/LinearBoltzmannSolver/lbs_compute_precursors.cc

```diff
@@ -0,0 +1,61 @@
+#include "lbs_linear_boltzmann_solver.h"
+
+#include "ChiMath/SpatialDiscretization/FiniteElement/PiecewiseLinear/pwl.h"
+
+#include "chi_log.h"
+extern ChiLog& chi_log;
+
+
+//###################################################################
+/**Compute the steady state delayed neutron precursor concentrations.*/
+void LinearBoltzmann::Solver::ComputePrecursors()
+{
+  auto pwl =
+      std::dynamic_pointer_cast<SpatialDiscretization_FE>(discretization);
+  const size_t J = max_precursors_per_material;
+
+  precursor_new_local.assign(precursor_new_local.size(), 0.0);
+
+  //================================================== Loop over cells
+  for (auto& cell : grid->local_cells)
+  {
+    const auto& fe_values = pwl->GetUnitIntegrals(cell);
+    const auto& transport_view = cell_transport_views[cell.local_id];
+    const double volume = transport_view.Volume();
+
+    //==================== Obtain xs
+    int cell_matid = cell.material_id;
+    int xs_id = matid_to_xs_map[cell_matid];
+
+    if ((xs_id < 0) || (xs_id >= material_xs.size()))
+    {
+      chi_log.Log(LOG_ALLERROR)
+          << "Cross-section lookup error\n";
+      exit(EXIT_FAILURE);
+    }
+
+    auto xs = material_xs[xs_id];
```

Comment:
Please change this to a const reference: `const auto& xs = *material_xs[xs_id]` and change the related arrow operators.

Reply:
Done.